lte/gateway/python/magma/pipelined/app/enforcement.py

```diff
@@ -496,4 +496,192 @@ def deactivate_rules(self, imsi, ip_addr, rule_ids):
             self._deactivate_flows_for_subscriber(imsi, ip_addr)
         else:
             for rule_id in rule_ids:
-                self._deactivate_flow_for_rule(imsi, ip_addr, rule_id)
\ No newline at end of file
+                self._deactivate_flow_for_rule(imsi, ip_addr, rule_id)
+
+    def _install_default_flow_for_session(self, seid):
+        """
+        Add a low priority flow to drop traffic in the event
+        that all rules have been deactivated.
+
+        Args:
+            seid (string): session id
+        """
+        match = MagmaMatch(imsi=seid)
+        actions = []  # empty options == drop
+        flows.add_drop_flow(self._datapath, self.tbl_num, match, actions,
+                            priority=self.ENFORCE_DROP_PRIORITY)
+
+    def _install_flow_for_qos_rules_session(self, seid, pdrs, qers):
+        """
+        Install a flow to get stats for a particular rule. Flows will match on
```

Comment:
is this docstring from another func?

Reply:
I corrected the all descriptions of the function.